File: backend/internal/models/member.go
package models

import (
	"database/sql/driver"
	"encoding/json"
	"time"
)

// Member 团队成员模型
type Member struct {
	ID         uint64    `gorm:"primaryKey;autoIncrement" json:"id"`
	Name       string    `gorm:"size:100;not null" json:"name"`
	Email      *string   `gorm:"size:255;uniqueIndex:uk_email" json:"email,omitempty"`
	Avatar     *string   `gorm:"size:500" json:"avatar,omitempty"`
	Role       string    `gorm:"size:50;not null;default:member;index:idx_role" json:"role"` // admin, leader, member
	Department *string   `gorm:"size:100;index:idx_department" json:"department,omitempty"`
	Skills     StringArray `gorm:"size:500;type:json" json:"skills"`
	Status     string    `gorm:"size:20;not null;default:active;index:idx_status" json:"status"` // active, inactive
	CreatedAt  time.Time `gorm:"autoCreateTime" json:"createdAt"`
	UpdatedAt  time.Time `gorm:"autoUpdateTime" json:"updatedAt"`

	// 关联
	Assignments       []Assignment       `gorm:"foreignKey:MemberID" json:"assignments,omitempty"`
	SubtaskAssignments []SubtaskAssignment `gorm:"foreignKey:MemberID" json:"subtaskAssignments,omitempty"`
}

// TableName 指定表名
func (Member) TableName() string {
	return "task_member"
}

// StringArray 自定义类型用于存储字符串数组
type StringArray []string

// Value 实现 driver.Valuer 接口
func (s StringArray) Value() (driver.Value, error) {
	if s == nil {
		return nil, nil
	}
	return json.Marshal(s)
}

// Scan 实现 sql.Scanner 接口
func (s *StringArray) Scan(value interface{}) error {
	if value == nil {
		*s = nil
		return nil
	}
	bytes, ok := value.([]byte)
	if !ok {
		return nil
	}
	return json.Unmarshal(bytes, s)
}

File: frontend/src/typings/api/member.ts
/** 成员角色 */
export type MemberRole = 'admin' | 'leader' | 'member';

/** 成员状态 */
export type MemberStatus = 'active' | 'inactive';

/** 成员信息 */
export interface Member {
  id: number;
  name: string;
  email?: string;
  avatar?: string;
  role: MemberRole;
  department?: string;
  skills: string[];
  status: MemberStatus;
  createdAt: string;
  updatedAt: string;
}

/** 成员创建请求 */
export interface MemberCreateRequest {
  name: string;
  email?: string;
  avatar?: string;
  role?: MemberRole;
  department?: string;
  skills?: string[];
  status?: MemberStatus;
}

/** 成员更新请求 */
export interface MemberUpdateRequest {
  name?: string;
  email?: string;
  avatar?: string;
  role?: MemberRole;
  department?: string;
  skills?: string[];
  status?: MemberStatus;
}

/** 成员列表筛选参数 */
export interface MemberListParams {
  status?: MemberStatus;
  role?: MemberRole;
  department?: string;
  keyword?: string;
  page?: number;
  pageSize?: number;
}

/** 分页成员列表响应 */
export interface MemberListResponse {
  members: Member[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

/** 成员统计 */
export interface MemberStatistics {
  total: number;
  active: number;
  inactive: number;
  admin: number;
  leader: number;
  member: number;
  departments: Record<string, number>;
}

/** 成员角色选项 */
export const MEMBER_ROLE_OPTIONS = [
  { label: '管理员', value: 'admin' },
  { label: '组长', value: 'leader' },
  { label: '成员', value: 'member' }
] as const;

/** 成员状态选项 */
export const MEMBER_STATUS_OPTIONS = [
  { label: '活跃', value: 'active' },
  { label: '停用', value: 'inactive' }
] as const;

/** 角色显示名称映射 */
export const MEMBER_ROLE_LABELS: Record<MemberRole, string> = {
  admin: '管理员',
  leader: '组长',
  member: '成员'
};

/** 状态显示名称映射 */
export const MEMBER_STATUS_LABELS: Record<MemberStatus, string> = {
  active: '活跃',
  inactive: '停用'
};
